report_migration/report_data.sql

```diff
@@ -0,0 +1,131 @@
+-- Step 1: Create temp_batch table
+DROP TEMPORARY TABLE IF EXISTS temp_batch;
+
+CREATE TEMPORARY TABLE IF NOT EXISTS temp_batch (
+    ID BIGINT NOT NULL,
+    created_at TIMESTAMP,
+    reportedID INT,
+    reportingID INT,
+    region_id INT,
+    x_coord INT,
+    y_coord INT,
+    z_coord INT,
+    timestamp TIMESTAMP,
+    manual_detect TINYINT,
+    on_members_world INT,
+    on_pvp_world TINYINT,
+    world_number INT,
+    equip_head_id INT,
+    equip_amulet_id INT,
+    equip_torso_id INT,
+    equip_legs_id INT,
+    equip_boots_id INT,
+    equip_cape_id INT,
+    equip_hands_id INT,
+    equip_weapon_id INT,
+    equip_shield_id INT,
+    equip_ge_value BIGINT,
+    report_sighting_id INT DEFAULT NULL,
+    report_location_id INT DEFAULT NULL,
+    report_gear_id INT DEFAULT NULL
+);
+
+-- Step 2: Insert data into temp_batch (manually adjust startId, endId, and batch_size as needed)
+
+INSERT INTO temp_batch (ID, created_at, reportedID, reportingID, region_id, x_coord, y_coord, z_coord,
+                        timestamp, manual_detect, on_members_world, on_pvp_world, world_number,
+                        equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id, equip_boots_id,
+                        equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id, equip_ge_value)
+SELECT ID, created_at, reportedID, reportingID, region_id, x_coord, y_coord, z_coord,
+       timestamp, manual_detect, on_members_world, on_pvp_world, world_number,
+       equip_head_id, equip_amulet_id, equip_torso_id, equip_legs_id, equip_boots_id,
+       equip_cape_id, equip_hands_id, equip_weapon_id, equip_shield_id, equip_ge_value
+FROM Reports
+WHERE ID > :startId AND ID <= :endId -- Python Variables
+ORDER BY ID ASC;
+
+-- Step 3: Update temp_batch with existing foreign keys from report_sighting
+UPDATE temp_batch AS tb
+LEFT JOIN report_sighting AS rs
+ON tb.reportingID = rs.reporting_id AND tb.reportedID = rs.reported_id AND tb.manual_detect = rs.manual_detect
+SET tb.report_sighting_id = rs.report_sighting_id;
+
+
+
+-- Debug: Check temp_batch after update
+-- SELECT * FROM temp_batch WHERE report_sighting_id IS NULL;
+
+-- Step 4: Insert missing rows into report_sighting
+INSERT INTO report_sighting (reporting_id, reported_id, manual_detect)
+SELECT DISTINCT reportingID, reportedID, manual_detect
+FROM temp_batch
+WHERE report_sighting_id IS NULL
+ORDER BY reportingID, reportedID, manual_detect;
+
+-- Debug: Check rows inserted into report_sighting
+-- SELECT * FROM report_sighting ORDER BY report_sighting_id DESC LIMIT 10;
+
+-- Step 5: Update temp_batch with newly inserted report_sighting IDs
+UPDATE temp_batch AS tb
+JOIN report_sighting AS rs
+ON tb.reportingID = rs.reporting_id AND tb.reportedID = rs.reported_id AND tb.manual_detect = rs.manual_detect
+SET tb.report_sighting_id = rs.report_sighting_id
+WHERE tb.report_sighting_id IS NULL;
```

Comment:
Any reason you did not adopt this from migrate_report_sighting
```sql
INSERT INTO report_sighting (reporting_id, reported_id, manual_detect)
SELECT DISTINCT tb.reportingID , tb.reportedID , IFNULL(tb..manual_detect,0) from temp_batch tb
WHERE 1
    AND NOT EXISTS (
        SELECT 1 FROM report_sighting rs
        WHERE 1
            AND tb.reportingID = rs.reporting_id
            AND tb.reportedID = rs.reported_id
            AND IFNULL(tb.manual_detect,0) = rs.manual_detect
);
```

Reply:
Step 3 finds existing report_sighting_id and sets it to the temp_batch 

so when step 4 mentions "WHERE report_sighting_id IS NULL" it does the same as 
"AND NOT EXISTS (
        SELECT 1 FROM report_sighting rs
        WHERE 1
            AND tb.reportingID = rs.reporting_id
            AND tb.reportedID = rs.reported_id
            AND IFNULL(tb.manual_detect,0) = rs.manual_detect
);"

Reply:
yes you are correct, but is there any particular reason you went for two updates?